src/api/ledger/orderbook.js

```diff
@@ -1,171 +1,54 @@
 'use strict';
 const _ = require('lodash');
+const async = require('async');
 const utils = require('./utils');
-const ripple = utils.common.core;
+const parseOrderbookOrder = require('./parse/orderbook-order');
 const validate = utils.common.validate;
-const bignum = require('bignumber.js');
+const composeAsync = utils.common.composeAsync;
+
+// account is to specify a "perspective", which affects which unfunded offers
+// are returned
+function getBookOffers(remote, account, ledgerVersion, limit,
+    takerGets, takerPays, callback) {
+  remote.requestBookOffers({
+    taker_gets: takerGets,
+    taker_pays: takerPays,
+    ledger: ledgerVersion || 'validated',
+    limit: limit,
+    taker: account
+  }, composeAsync((data) => data.offers.map(parseOrderbookOrder), callback));
+}
+
+function isSameIssue(a, b) {
+  return a.currency === b.currency && a.counterparty === b.counterparty;
+}
+
+function orderFilter(issue, direction, order) {
+  return isSameIssue(issue, order.specification.quantity)
+    && order.specification.direction === direction;
+}
+
+function formatBidsAndAsks(orderbook, orders) {
+  // the "base" currency is the currency that you are buying or selling
+  // the "counter" is the currency that the "base" is priced in
+  // a "bid"/"ask" is an order to buy/sell the base, respectively
+  const bids = orders.filter(_.partial(orderFilter, orderbook.base, 'buy'));
+  const asks = orders.filter(_.partial(orderFilter, orderbook.base, 'sell'));
```

Comment:
There was something about this is nagging at me about being off. I can't quite articulate it. I can say `bids.length + asks.length !== orders.length` if that alerts you to anything :)